backend/matching-service/src/services/rabbitmq.service.ts

```diff
@@ -0,0 +1,82 @@
+import client, { Connection, Channel } from 'amqplib'
+
+import config from '../common/config.util'
+import { IUserQueueMessage } from '../types/IUserQueueMessage'
+import logger from '../common/logger.util'
+
+class RabbitMQConnection {
+    connection!: Connection
+    channel!: Channel
+    private connected!: boolean
+
+    async connect() {
+        if (this.connected && this.channel) return
+        else this.connected = true
+
+        try {
+            logger.info(`[Init] Connecting to Rabbit-MQ Server`)
+            this.connection = await client.connect({
+                protocol: 'amqp',
+                hostname: config.RMQ_HOST,
+                port: 5672,
+                username: config.RMQ_USER,
+                password: config.RMQ_PASSWORD,
+            })
+
+            logger.info(`[Init] Rabbit MQ Connection is ready`)
+
+            this.channel = await this.connection.createChannel()
+
+            logger.info(`[Init] Created RabbitMQ Channel successfully`)
+        } catch (error) {
+            logger.error(error)
+            logger.error(`[Error] Not connected to MQ Server`)
+        }
+    }
+
+    // Direct exchange for entry queue
+    async sendToEntryQueue(message: IUserQueueMessage) {
+        try {
+            if (!this.channel) {
+                await this.connect()
+            }
+
+            // Set durable true to ensure queue stays even with mq restart (does not include message persistance)
+            await this.channel.assertExchange('Entry-Queue', 'direct', { durable: false })
+
+            await this.channel.assertQueue('entry_queue', { durable: false })
+
+            // Params: Queue, Exchange, Routing Key
+            await this.channel.bindQueue('entry_queue', 'Entry-Queue', 'entry')
+            this.channel.publish('Entry-Queue', 'entry', Buffer.from(JSON.stringify(message)))
+        } catch (error) {
+            console.error(error)
```

Comment:
can remove this try catch, the default exception handler will log this error

Reply:
are you referring to the errorHandler.middleware.ts?

Reply:
yes, you can catch here and log it here as well if u wanna have more localised error messages

Reply:
I thought the errorHandler is for express, and not for anything else? Since it is only used within app.use(defaultErrorHandler)

Reply:
ur calling this in the context of a request coming in thru express to add a user to the queue right? the call stack is extended from an initial express request

Reply:
I see, but in the bottom context, entryConsumer is independent and not called thru express. Does that mean i remove this try catch and keep the bottom one?

Reply:
in that case then yeah, probs need a seperate try catch

Reply:
would it be better to standardise this within rabbitmq? As only the sendToEntryQueue is called with express and the rest are independent

Reply:
you can make the call, i would choose the route of less coupling